### PR TITLE
CAA records

### PR DESCRIPTION
**Pull request by yannik-b**
Add support for a CAA record for a whole domain.

CAA record is being used to specify a CA for this domain.
Only this CA can create certificates for the domain.

Also see: https://www.rfc-editor.org/rfc/rfc8659.html

### CODE REVIEW

**Comment by t2d** [comment]
Thank you for the PR. Please put a for loop in there as it is possible to have multiple CAA records for a domain. Thanks!

**Comment by yannik-b** [comment]
Hi, I thought it is only possible to have one CAA record per domain so I removed the for loop. But you're right, one can provide multiple records.
Now it loops over the records.

File: templates/bind/zones/db.template.j2 [hunk 1]
```diff
@@ -52,6 +54,14 @@ $TTL {{ zone.negative_ttl|default('3600') }}   ; 1 hour
                     TXT     "{{ txt }}"
 {%   endfor %}
 {% endif %}
+{# CAA record is being used to specify a CA for this domain.
+   Only this CA can create certificates for the domain.
+   Also see: https://www.rfc-editor.org/rfc/rfc8659.html #}
+{% if zone.caa_records is defined %}
+{%   for caa in zone.caa_records %}
+                    CAA     {{ caa }}
+{%   endfor %}
```
Comment (t2d) [inline templates/bind/zones/db.template.j2:63]: The indentation is a bit unusual to me.  Are you okay with spaces before the expression?

```
  {% for caa in zone.caa_records %}
                    CAA     {{ caa }}
  {% endfor %}
```

**Comment by t2d** [comment]
@yannik-b Do you still want this merged?

**Comment by t2d** [comment]
@yannik-b thank you for your PR and sorry for not looking carefully into it

**Pull request merged**